stable-baselines3/sb3.py

```diff
@@ -66,33 +67,30 @@ def _on_rollout_start(self):
 
         self.n_updates += 1
 
-    def _evaluate_model(self) -> float:
-        """
-        Evaluate the current model on the evaluation environment.
-
-        :return: The mean reward obtained by the model.
-        """
-
-        if self.verbose > 0:
-            print("Starting evaluation of model")
-
-        episode_rewards = []
-        for _ in range(self.eval_env.num_envs):
-            obs = self.eval_env.reset()
-            done = False
-            episode_reward = 0.0
-            while not done:
-                action, _ = self.model.predict(obs, deterministic=True)
-                obs, reward, done, _ = self.eval_env.step(action)
-                episode_reward += reward
-            episode_rewards.append(episode_reward)
 
-        mean_reward = np.mean(episode_rewards)
+def evaluate_model(env, model, num_episodes=1, deterministic=True):
+    """
+    Run the model on the given environment either for a number of episodes or until a stopping condition is met.
 
-        if self.verbose > 0:
-            print(f"Evaluation complete: mean reward {mean_reward}")
+    :param env: Environment to run the model in.
+    :param model: The trained model to use for predictions.
+    :param num_episodes: Number of episodes to run.
```

Comment:
I do like the number of episodes to run idea. It wouldn't be necessary if deterministic == true, however. What if the method assumed a single run of the environment?

Reply:
Also, there is a way to step through all the environments in parallel. This is what inference loop does, but the callback loop does not and instead goes sequentially through the environments.